lib/device-log/ios-log.js

```diff
@@ -25,20 +26,46 @@ class IOSLog {
   }
 
   async startCaptureRealDevice () {
-    let spawnEnv = _.clone(process.env);
-    logger.debug('Attempting iOS device log capture via libimobiledevice idevicesyslog');
-    try {
-      let idevicesyslog = await fs.which('idevicesyslog');
-      logger.debug(`Found idevicesyslog: '${idevicesyslog}'`);
-      this.proc = new SubProcess('idevicesyslog', ['-u', this.udid], {env: spawnEnv});
-    } catch (err) {
-      logger.warn('Could not capture device log using libimobiledevice idevicesyslog. ' +
-                  'Libimobiledevice is probably not installed');
+    if ((this.realDeviceLogger || '').indexOf('idevicesyslog') !== -1) {
+      logger.debug('Attempting iOS device log capture via libimobiledevice idevicesyslog');
+      let idevicesyslog;
+      try {
+        if (this.realDeviceLogger.toLowerCase() === 'idevicesyslog') {
+          idevicesyslog = await fs.which('idevicesyslog');
+        } else {
+          idevicesyslog = this.realDeviceLogger;
+        }
+        logger.debug(`Using idevicesyslog: '${idevicesyslog}'`);
+
+        this.proc = new SubProcess('idevicesyslog', ['-u', this.udid], {env: process.env});
+      } catch (err) {
+        logger.errorAndThrow(`Unable to capture device log using idevicesyslog: ${err.message}`);
+      }
+    } else if ((this.realDeviceLogger || '').indexOf('deviceconsole') !== -1) {
       logger.debug('Attempting iOS device log capture via deviceconsole');
-      spawnEnv.PATH = `${process.env.PATH}:${DEVICE_CONSOLE_PATH}`;
-      spawnEnv.DYLD_LIBRARY_PATH = `${DEVICE_CONSOLE_PATH}:${process.env.DYLD_LIBRARY_PATH}`;
-      this.proc = new SubProcess('deviceconsole', ['-u', this.udid], {env: spawnEnv});
+      let deviceconsole;
+      if (this.realDeviceLogger.toLowerCase() === 'deviceconsole') {
+        deviceconsole = DEVICE_CONSOLE_PATH;
+      } else {
+        deviceconsole = this.realDeviceLogger;
+      }
+      logger.debug(`Using default deviceconsole: '${deviceconsole}'`);
+
+      try {
+        // first make sure the executable is actually there
+        await fs.which(path.resolve(deviceconsole, 'deviceconsole'));
```

Comment:
do we expect the user to know that they should not send in the path of deviceconsole itself but rather its containing directory? i imagine they will probably send in the path of deviceconsole itself...